tests/test_olio_grid_functions.py

```diff
@@ -0,0 +1,236 @@
+    ((1, 0), (0, 1), np.array([[False, True], [True, True]], dtype = bool)),
+    ((1, 0), (1, 2), np.array([[False, False], [True, True]], dtype = bool)),
+    ((1, 0), (2, 1), np.array([[False, False], [True, False]], dtype = bool)),
+    ((1, 2), (0, 1), np.array([[False, True], [False, False]], dtype = bool)),
+    ((1, 2), (1, 0), np.array([[True, True], [False, False]], dtype = bool)),
+    ((1, 2), (2, 1), np.array([[True, True], [True, False]], dtype = bool)),
+    ((0, 1), (1, 0), np.array([[True, False], [False, False]], dtype = bool)),
+    ((0, 1), (2, 1), np.array([[True, False], [True, False]], dtype = bool)),
+    ((0, 1), (1, 2), np.array([[True, False], [True, True]], dtype = bool)),
+    ((2, 1), (1, 0), np.array([[True, True], [False, True]], dtype = bool)),
+    ((2, 1), (0, 1), np.array([[False, True], [False, True]], dtype = bool)),
+    ((2, 1), (1, 2), np.array([[False, False], [False, True]], dtype = bool)),
+])
+def test_left_right_foursome(previous, next, expected_result):
+
+    # --------- Arrange ----------
+    full_p_list = [previous, (1, 1), next]
+    p_index = 1
+    # --------- Act ----------
+    result = left_right_foursome(full_pillar_list = full_p_list, p_index = p_index)
+    # --------- Assert ----------
+    np.testing.assert_equal(result, expected_result)
+
+
+def test_left_right_foursome_error_handling():
+
+    # --------- Arrange ----------
+    p_index = 1
+    # arrange for exceptions
+    full_p_list_entry_error = [(1, 0), (1, 2), (0, 1)]
+    full_p_list_exit_error = [(1, 0), (1, 1), (2, 2)]
+
+    # --------- Act and Assert----------
+    with pytest.raises(Exception) as excinfo:
+        result_entry_error = left_right_foursome(full_pillar_list = full_p_list_entry_error, p_index = p_index)
+        assert "code failure whilst taking entry sides from dubious full pillar list" in str(excinfo.value)
+    with pytest.raises(Exception) as excinfo2:
+        result_exit_error = left_right_foursome(full_pillar_list = full_p_list_exit_error, p_index = p_index)
+        assert "code failure whilst taking exit sides from dubious full pillar list" in str(excinfo.value)
+
+
+def test_resequence_nexus_corp():
+
+    # TODO: confirm you used the correct cornerpoint ordering from Nexus (pg 1424 Nexus Keyword file)
+    # --------- Arrange ----------
+    corner_points = np.array([[[[[[[0., 0., 0.], [10., 0., 0.]], [[10., -10., 0.], [0., -10., 0.]]],
+                                 [[[0., 0., 10.], [10., 0., 10.]], [[10., -10., 10.], [0., -10., 10.]]]]]]])
+
+    expected_result = np.array([[[[[[[0., 0., 0.], [10., 0., 0.]], [[0., -10., 0.], [10., -10., 0.]]],
+                                   [[[0., 0., 10.], [10., 0., 10.]], [[0., -10., 10.], [10., -10., 10.]]]]]]])
+
+    # --------- Act ----------
+    resequence_nexus_corp(corner_points = corner_points)
+
+    # --------- Assert ----------
+    np.testing.assert_almost_equal(corner_points, expected_result)
+
+
+# TODO: Find out if grid is too simple
```

Comment:
To make a significant test of the function, a complex geometry would be needed, with some cells pinched out and others collapsed in either the I or J direction, or any combination thereof.  At least this test will execute the function, so could pick up any basic errors not detected by flake8.

Reply:
Hi @andy-beer,

I pinched out the entire layer and the expected cells are not returned by the random_cell function. I am clearly doing something wrong when I try to collapse one cell in the j direction, as that cellkji0 is still returned. This is an official SOS!

Reply:
As indicated in code comments, I believe the issue is a misplaced write_hdf5().  I will make the correction.